debian/changelog

```diff
@@ -1,3 +1,17 @@
+cloud-init (22.3.3-0ubuntu1~22.10.1) kinetic; urgency=medium
```

Comment:
I think we have two choices here to maintain upgrade path on kinetic to next new-upstream-snapshot:

  1. manually set this packaged release version `22.3-19....` so that any subsequent `new-upstream-snapshots` before we publish `22.4`  published to kinetic are > `22.3.3`
  2. add a commit to main to set version.py to 22.3.3 so any subsequent new-upstream-snapshot into kinetic will be `22.3.3-X...`

If we don't do either, our next new-upstream-snapshot into kinetic will be:
`22.3-46-g9aa4ec69-0ubuntu1~22.10.1`  which will be less than a published `22.3.3-0ubuntu1~22.10.1`

This version comparison will break deb package upgrade path logic and not allow us to upload such a version to PPAs or -updates/-proposed pockets.

Reply:
I lean away from manually setting versions if there are alternative options, though I'm not sure how the other option would work.

Reply:
If we desire to be able to snapshot main into ubuntu/devel but then update to, for example, a package built from 22.3.4, then I think that the easiest way might be to bump `ubuntu1` to `ubuntu2` for the snapshots from main. That will ensure that we can always "update" back to the next release version. It is worth noting that this version scheme is a little misleading, however, because doing a snapshot from main and leaving the version starting with `22.3.3` implies that the package is based on `22.3.3` when it is not. Maybe we don't care about that for devel?

On the contrary, if we don't believe that we'll ever want to do a snapshot from main followed by a version bump back down to a minor release (22.3.4), then just pre-releasing the devel package under the next release name `22.4-<commit number>` makes sense I think, since main is the branch that the next release will be based on, and number of commits will always increase so I don't think there should ever be an upgrade issue.

I'm fine to just manually bump for now if that's what you'd prefer, but I'd like to think that with this versioning issue that we can remove the human from the loop.

Thoughts?

Reply:
My above comment ignores how our current tooling works and just tries to make sense of upgrade paths and versions for ubuntu/devel.

Reply:
Ok we decided as a squad in standup we would like to preserve upstream release prefixes in our deb packaging releases in Ubuntu so the version prefix will be 22.3.3-0ubuntu1~22.10.1.   Any interim releases which land in future devel series will likely have package major.minor version prefix related to the next release under development, but a deminished version string including `~pre.#-` to ensure the interim release version `22.4~pre.x-0ubuntu1~23.04.1` sorts lower than the official upstream release version of `22.4-ubuntu1~23.04.1`